source3/smbd/trans2.c

```diff
@@ -229,8 +229,17 @@ NTSTATUS get_ea_value(TALLOC_CTX *mem_ctx,
 				ea_name, val, attr_size);
 	}
 
-	if (sizeret == -1 && errno == ERANGE && attr_size != 65536) {
-		attr_size = 65536;
+	#define ALLOW_LARGE_XA 1
```

Comment:
Although CPP doesn't take the C code blocks into account, I'd suggest to put this #define/#if blocks outside of the function definition, as you use this macro value later in the code.

to the line 204, for example.

Reply:
On line 214: I'd suggest to raise initial attribute value size to 1K:
```
size_t attr_size = 1024;
```
So most of the common attributes would fit into that buffer from the first attempt. The leap between 256 bytes and 2Mb is quite big and it could be reasonable to have some intermediate value tried before going so extreme. The same 64K could be an alternative.

Reply:
On line 222: It could be reasonable to add DEBUG() line here as well, so we can see in the logs that allocating 2M chunks of memory create any problems.